pkg/image/controller/controller.go

```diff
@@ -55,9 +217,6 @@ const retryCount = 2
 // 4. ImageStreamMapping save error
 // 5. error when marking ImageStream as imported
 func (c *ImportController) Next(stream *api.ImageStream) error {
```

Comment:
> Hmm, it just occurred to me that we need to make sure this function is thread safe. Initializing the shared client here isn't, and I need to sweep getTags/importTags and the client functions to see if it is building connection caches in a thread safe way.

I agree the thread safety is important, but since client is only "injected for testing" per the godoc (and verified in the code), that client looks local to me.  So it results in one per call, so the connection establishment below should be safe, barring insanity.